src/lib/operators/jit_operator/operators/jit_read_tuple.cpp

```diff
@@ -0,0 +1,117 @@
+#include "jit_read_tuple.hpp"
+
+#include "constant_mappings.hpp"
+#include "resolve_type.hpp"
+#include "storage/create_iterable_from_column.hpp"
+
+namespace opossum {
+
+std::string JitReadTuple::description() const {
+  std::stringstream desc;
+  desc << "[ReadTuple] ";
+  for (const auto& input_column : _input_columns) {
+    desc << "x" << input_column.tuple_value.tuple_index() << " = Col#" << input_column.column_id << ", ";
+  }
+  for (const auto& input_literal : _input_literals) {
+    desc << "x" << input_literal.tuple_value.tuple_index() << " = " << input_literal.value << ", ";
+  }
+  return desc.str();
+}
+
+void JitReadTuple::before_query(const Table& in_table, JitRuntimeContext& ctx) {
+  // Create a runtime tuple of the appropriate size
+  ctx.tuple.resize(_num_tuple_values);
+
+  // Copy all input literals to the runtime tuple
+  for (const auto& input_literal : _input_literals) {
+    auto data_type = input_literal.tuple_value.data_type();
+    resolve_data_type(data_type, [&](auto type) {
+      using DataType = typename decltype(type)::type;
+      input_literal.tuple_value.materialize(ctx).set<DataType>(boost::get<DataType>(input_literal.value));
+    });
+  }
+
+  // Create a JitColumnReader for each input column. We use the first chunk in the table to determine the column type.
+  for (const auto& input_column : _input_columns) {
+    const auto column_id = input_column.column_id;
+    const auto column = in_table.get_chunk(opossum::ChunkID{0})->get_column(column_id);
```

Comment:
@mrzzzrm - This is the old "column in a table vs column in a chunk" naming discussion again :(

Reply:
Not longer relevant, since the code has changed and we no longer need to access the first chunk.